### DISCUSSION TITLE
[client-preset] Opt out of graphql function?

### DISCUSSION THREAD

**Discussion by aaronadamsCA** [dotansimha/graphql-code-generator#9901] [Ideas]
We migrated to the client preset, but the way we're using it feels much simpler and more performant than what the docs suggest.

- Our GraphQL operations and fragments live in `.graphql` files only.
- Our `graphql.config.ts` reads `.graphql` files only.
- Our components import operation types and document nodes from the generated `graphql.ts` file.

This works really well, with far fewer tradeoffs than the standard setup.

At the very least, I think this should be officially supported. It's probably too late to make the `graphql` function opt-in, but could it at least be opt-out? We don't need all that extra code generated; we're never going to use it.

I think a truly ideal future state would have the client preset always generate exactly one file, with the user able to `presetConfig` its exports. You could choose from among operation types, document nodes, fragment masking, and a `graphql()` function, and the generated file would contain only the necessary code to provide the exports you requested.

I think this would continue to align well with the intentions of the client preset, while providing the opportunity to greatly improve performance in larger projects.

**Comment by manuelmeister** [comment]
@aaronadamsCA can you show an example config?

**Reply by aaronadamsCA** [reply]
I can provide an example repository that I think works well! I am currently on deadline though so I'll need some time, I'll follow up once I have it.
